### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by arahlin**
This PR contains several changes for cleaning up internal attribute handling, renaming various user options for clarity/consistency, and some reorganization of user-facing functions for handling data.

### CODE REVIEW

File: xfaster/xfaster_class.py [hunk 1]
```diff
@@ -511,32 +496,87 @@ def __del__(self):
             finally:
                 handler.release()
 
+    def _get_sim_files(
+        self, name, root=None, subset="*", suffix="", data_suffix="", fs=None
+    ):
+        """
+        Convenience function for finding all matching sims per map
```
Comment (annegambrel) [inline xfaster/xfaster_class.py:503]: need docstring for arguments-- would be good to include example args from the example script

**Review by annegambrel — COMMENTED** [review]
This is great-- I think you're still working on changes but let me know what testing you've done/if it's in a good state for me to run this on signal and nulls to see where it breaks!

**Comment by arahlin** [comment]
> This is great-- I think you're still working on changes but let me know what testing you've done/if it's in a good state for me to run this on signal and nulls to see where it breaks!

I’ve run this on the example script locally and on the tutorial notebook.  Would be great to run this on some other dataset for sure, especially to test template subtraction and Planck signal subtraction.

I’m working on two things:

* Combining `get_masked_data`, `get_masked_fake_data` and the data pieces in `get_masked_sims` into a single function
* Reconfiguring the Planck things to use more general terminology, e.g. `reference_signal` subtraction and A/B reference halves for templates.

A longer term project is reconfiguring the likelihood code to make it easier to plug into other solvers, e.g. Cobaya.  That will be a separate PR though.

File: xfaster/xfaster_class.py (hunk 1 above)
Comment (arahlin) [reply xfaster/xfaster_class.py:503]: This is an internal function that isn't meant to be exposed to the user, so I don't think that a complete docstring with examples is necessary.  I can document the list of arguments, though.

Comment (annegambrel) [reply xfaster/xfaster_class.py]: Ah, makes sense. I just found myself puzzling about what the arguments were and think some examples could be helpful if anyone ever has to add to this/debug it.

**Comment by annegambrel** [comment]
> > This is great-- I think you're still working on changes but let me know what testing you've done/if it's in a good state for me to run this on signal and nulls to see where it breaks!
> 
> I’ve run this on the example script locally and on the tutorial notebook. Would be great to run this on some other dataset for sure, especially to test template subtraction and Planck signal subtraction.
> 
> I’m working on two things:
> 
> * Combining `get_masked_data`, `get_masked_fake_data` and the data pieces in `get_masked_sims` into a single function
> * Reconfiguring the Planck things to use more general terminology, e.g. `reference_signal` subtraction and A/B reference halves for templates.
> 
> A longer term project is reconfiguring the likelihood code to make it easier to plug into other solvers, e.g. Cobaya. That will be a separate PR though.

Given that there are more significant changes coming I'll hold off on a barrage of testing unless I find myself with some free time.

**Comment by arahlin** [comment]
Ok, I think most of the code changes are in place now, and tested on the tutorial notebook and example script.  Still need to go through the tutorial notebook and make sure that all the text there reflects the changes properly.

**Comment by arahlin** [comment]
I'm hands off on this now!  Any other changes I can think of should be separate PRs for later.

**Pull request merged**